preload/resources/fetch-destination-worker.js

```diff
@@ -0,0 +1,11 @@
+self.addEventListener('fetch', function(event) {
+    if (event.request.url.indexOf('dummy.xml') != -1) {
+        var response = new Response();
+        if (event.request.destination == "")
+            response.status = 200;
+        else
+            response.status = 500;
```

Comment:
This doesn't match the Preload specification. It doesn't say that an error event is dispatched for such status codes. That would also likely be problematic due to the same-origin policy mostly trying to keep that a secret.